cli/cmd/vuln_host.go

```diff
@@ -493,93 +510,141 @@ func hostVulnSummaryFromHostDetail(hostVulnSummary *api.HostVulnCveSummary) (str
 				continue
 			}
 
-			added := false
-			for i := range out {
-				if out[i][1] == strings.Title(pkg.Severity) &&
-					out[i][2] == pkg.Name &&
-					out[i][3] == pkg.Version &&
-					out[i][4] == pkg.FixedVersion &&
-					out[i][5] == pkg.PackageStatus {
-
-					if countCVEs, err := strconv.Atoi(out[i][0]); err == nil {
-						out[i][0] = fmt.Sprintf("%d", (countCVEs + 1))
-						added = true
-					}
-
-					if withHosts {
-						if countHosts, err := strconv.Atoi(out[i][6]); err == nil {
-							prevCount := stringToInt(pkg.HostCount)
-							out[i][6] = fmt.Sprintf("%d", (countHosts + prevCount))
-							added = true
-						}
-					}
-
+			if vulCmdState.Severity != "" {
+				if filterSeverity(pkg.Severity, vulCmdState.Severity) {
+					filteredPackages = aggregatePackagesWithHosts(filteredPackages, pack, withHosts)
+					continue
 				}
 			}
+			aggregatedPackages = aggregatePackagesWithHosts(aggregatedPackages, pack, withHosts)
+		}
+	}
 
-			if added {
-				continue
-			}
-
-			row := []string{
-				"1",
-				strings.Title(pkg.Severity),
-				pkg.Name,
-				pkg.Version,
-				pkg.FixedVersion,
-				pkg.PackageStatus,
-			}
-			if withHosts {
-				row = append(row, pkg.HostCount)
-			}
-			out = append(out, row)
+	for _, p := range aggregatedPackages {
+		output := []string{
+			strconv.Itoa(p.cveCount),
+			p.severity,
+			p.packageName,
+			p.currentVersion,
+			p.fixVersion,
+			p.packageStatus}
+		if p.hostCount > 0 {
+			output = append(output, strconv.Itoa(p.hostCount))
 		}
+		out = append(out, output)
 	}
 
 	// order by severity
 	sort.Slice(out, func(i, j int) bool {
 		return severityOrder(out[i][1]) < severityOrder(out[j][1])
 	})
 
-	return out
+	if len(filteredPackages) > 0 {
+		filteredOutput := fmt.Sprintf("%v of %v package(s) showing \n", len(out), len(aggregatedPackages)+len(filteredPackages))
+		return out, filteredOutput
+	}
+
+	return out, ""
 }
 
-func hostVulnCVEsTable(cves []api.HostVulnCVE) [][]string {
-	out := [][]string{}
-	out = append(out, hostVulnCVEsTableForSeverity(cves, "Critical")...)
-	out = append(out, hostVulnCVEsTableForSeverity(cves, "High")...)
-	out = append(out, hostVulnCVEsTableForSeverity(cves, "Medium")...)
-	out = append(out, hostVulnCVEsTableForSeverity(cves, "Low")...)
-	//out = append(out, hostVulnCVEsTableForSeverity(cves, "Info")...)
-	out = append(out, hostVulnCVEsTableForSeverity(cves, "Negligible")...)
-	return out
+func hostVulnCVEsTable(cves []api.HostVulnCVE) ([][]string, string) {
+	var out [][]string
+	var filteredCves = 0
+	var totalCves = 0
+
+	for _, severity := range api.ValidEventSeverities {
+		_, sev := severityToProperTypes(severity)
+		output, filtered, total := hostVulnCVEsTableForSeverity(cves, sev)
+		filteredCves += filtered
+		totalCves += total
+		out = append(out, output...)
+	}
+
+	if filteredCves > 0 {
+		showing := totalCves - filteredCves
+		return out, fmt.Sprintf("\n %v of %v cve(s) showing \n", showing, totalCves)
```

Comment:
I think this should not have a space.... 🤔 maybe...
```suggestion
		return out, fmt.Sprintf("\n%v of %v cve(s) showing \n", showing, totalCves)
```